WalletWasabi.Fluent/Controls/CurrencyEntryBox.axaml.cs

```diff
@@ -255,10 +255,10 @@ private bool ValidateEntryText(string preComposedText)
 
 		protected override void OnKeyDown(KeyEventArgs e)
 		{
-			DoPasteCheckAsync(e);
+			DoPasteCheck(e);
 		}
 
-		private async void DoPasteCheckAsync(KeyEventArgs e)
+		private void DoPasteCheck(KeyEventArgs e)
```

Comment:
I would expect it to be:

```cs
		protected override void OnKeyDown(KeyEventArgs e)
		{
			var keymap = AvaloniaLocator.Current.GetService<PlatformHotkeyConfiguration>();

			bool Match(List<KeyGesture> gestures) => gestures.Any(g => g.Matches(e));

			if (Match(keymap.Paste))
			{
				ModifiedPasteAsync();
			}
			else
			{
				base.OnKeyDown(e);
			}
		}
```

i.e. no `DoPasteCheckAsync` at all. 

Could @jmacato (#5213) explain why `DoPasteCheckAsync` is needed? And why 

```cs
public async void ModifiedPasteAsync()
```

instead of

```cs
public async Task ModifiedPasteAsync()
```
?

Reply:
Probably even

```cs
		protected override void OnKeyDown(KeyEventArgs e)
		{
			var keymap = AvaloniaLocator.Current.GetService<PlatformHotkeyConfiguration>();

			bool Match(List<KeyGesture> gestures) => gestures.Any(g => g.Matches(e));

			if (Match(keymap.Paste))
			{
				_ = ModifiedPasteAsync();
			}
			else
			{
				base.OnKeyDown(e);
			}
		}

		public async Task ModifiedPasteAsync()
		{
			var text = await AvaloniaLocator.Current.GetService<IClipboard>().GetTextAsync();

			if (string.IsNullOrEmpty(text))
			{
				return;
			}

			text = text.Replace("\r", "").Replace("\n", "").Trim();

			// Based on broad M0 money supply figures (80 900 000 000 000.00 USD).
			// so USD has 14 whole places + the decimal point + 2 decimal places = 17 characters.
			// Bitcoin has "21 000 000 . 0000 0000".
			// Coincidentally the same character count as USD... weird.
			// Plus adding 4 characters for the group separators.
			if (text.Length > 17 + 4)
			{
				text = text[..(17 + 4)];
			}

			if (ValidateEntryText(text))
			{
				OnTextInput(new TextInputEventArgs { Text = text });
			}
		}
```

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/7000/files?diff=split&w=1#diff-06e212bf8992c4124ecbaaecf24f385eae47367ec94e0703b1cc7704ee2d2118R258 fixed here